src/app/components/map/map.component.ts

```diff
@@ -1647,30 +1648,131 @@ export class MapComponent implements AfterViewInit {
         break;
     }  
 
+
   }
 
-  indoorView(): void
+  /**
+   * This method is called when user presses "Enter building" button, and it shows a drop down menu and exit button
+   * which allows the user to view different floors in the building.
+   * @param buildingInfo is a dictionary that holds informations about the buildings
+   * @param polygon is the building layer
+   * @param marker is the building marker
+   */
+  indoorView(buildingInfo: any, polygon: any, marker: any): void
   {
-
-    var hallOverlay;
-
-    var imageBoundHall = {
-      north: 45.497735, //Top
-      south: 45.496807, //Bottom
-      east: -73.578316, //Right
-      west: -73.579586 //Left
+    let floorImage = ""; //Holds the image path
+    var indoorOverlay; //Layer on top of building
+    let self = this;
+    let empty = "";
+
+    var imageBound = {
+      north: buildingInfo["bound"].north, //Top
+      south: buildingInfo["bound"].south, //Bottom
+      east: buildingInfo["bound"].east, //Right
+      west: buildingInfo["bound"].west //Left
     };
-  
-    hallOverlay = new google.maps.GroundOverlay(
-        'assets/FloorImages/Hall/hall-8.png', 
-        imageBoundHall);
-        
-    hallOverlay.setMap(this.map);
+
+    indoorOverlay = new google.maps.GroundOverlay(
+        floorImage, 
+        imageBound);
+        indoorOverlay.setMap(this.map);
 
     //Zoom in
-    this.map.setCenter({lat: 45.497280, lng: -73.578940});
+    this.map.setCenter({lat: buildingInfo["Location"].lat, lng: buildingInfo["Location"].lng});
     this.map.setZoom(19);
+    //No zoom or drag anymore
+    this.map.setOptions({draggable: false, zoomControl: false, scrollwheel: false, disableDoubleClickZoom: true});
```

Comment:
Right now when the user presses the "enter building" button, it unables the user to zoom or drag when viewing the floors.
I can change this if we need to allow the user to zoom